sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -131,38 +146,34 @@ void setPackageDeclaration(String packageName) {
     void createClass(String serviceInterfaceImplShortName, String serviceInterfaceShortName,
         TemplateInput templateInput) {
         classBuilder = compilationUnit.addClass(serviceInterfaceImplShortName, Modifier.Keyword.PUBLIC);
-
         String serviceInterfacePackage = templateInput.getServiceInterfaceFQN()
             .substring(0, templateInput.getServiceInterfaceFQN().lastIndexOf('.'));
+
         compilationUnit.addImport(serviceInterfacePackage + "." + serviceInterfaceShortName);
         classBuilder.addImplementedType(serviceInterfaceShortName);
 
         configureLoggerField(classBuilder.addField("ClientLogger", "LOGGER", Modifier.Keyword.PRIVATE,
             Modifier.Keyword.STATIC, Modifier.Keyword.FINAL), serviceInterfaceShortName);
 
-        classBuilder.addField(HttpPipeline.class, "defaultPipeline", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
-        classBuilder.addField(ObjectSerializer.class, "serializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
-
+        classBuilder.addField(HttpPipeline.class, "httpPipeline", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
         compilationUnit.addImport(JsonSerializer.class);
+        classBuilder.addField(JsonSerializer.class, "jsonSerializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
+        compilationUnit.addImport(XmlSerializer.class);
+        classBuilder.addField(XmlSerializer.class, "xmlSerializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
```

Comment:
Long term we'll want to conditionally add in `JsonSerializer` and `XmlSerializer` based on the `Content-Type`s and `Accept`s being used in the requests being sent.

Reply:
We are suing them conditionally if thats what you meant.
Check the generated impl https://gist.github.com/samvaity/37483c22e6633719d62537c6b6e4a375